util/predictoor/query.py

```diff
@@ -0,0 +1,79 @@
+from typing import Dict, Optional, Tuple
+
+from util.graphutil import submitQuery
+from util.predictoor.models import Predictoor, Prediction
+
+
+def queryPredictoors(st_block: int, end_block: int, chainID: int):
+    """
+    @description
+        Queries the predictPredictions GraphQL endpoint for a given
+        range of blocks and chain ID, and returns a dictionary of
+        Predictoor objects, where each Predictoor object represents
+        a unique user who has made predictions during the given block range.
+
+    @params
+        st_block (int) -- The starting block number for the query.
+        end_block (int) -- The ending block number for the query.
+        chainID (int) -- The ID of the chain to query.
+
+    @return
+        predictoors -- A dictionary of address to Predictoor objects
+
+    @raises
+        AssertionError: If the result of the query contains an error.
+    """
+    predictoors: Dict[str, Predictoor] = {}
+
+    chunk_size = 1000
+    offset = 0
+
+    while True:
+        # pylint: disable=line-too-long
+        query = """
+        {
```

Comment:
I would suggest moving all the raw queries in the project to a separate file or module. However, this seems low priority. I suggest creating a ticket if you agree. Otherwise please mark this comment as resolved.

Reply:
> implementing casing standards as the project grows is beneficial for consistency

I'm ok if we create a new github issue for this. In df-py, I was using casing standards that no longer the standard in python, we really should switch to the proper ones (snake case for function names, etc). 

It could be something for Calina to do, to help learn the codebase.

> suggest moving all the raw queries in the project to a separate file or module

OK, though that was kinda the original point of query.py, hence its name. 

Better: 
- move most challenge DF-related queries into df-py/util/challenge/query.py. And other new modules from challenge (like I've already started).
- similarly, move predictoor DF-related queries into df-py/util/predictoor/query.py
- and to wrap it up, move volume DF-related queries into df-py/util/volume/query.py. And move other volume-specific DF modules there (many!)

Reply:
Re the new directories:
- for _this_ issue just move the new predictoor DF-related queries into df-util/predictoor/query.py
- and create a new issue for the rest

Reply:
> Re the new directories:
> 
> * for _this_ issue just move the new predictoor DF-related queries into df-util/predictoor/query.py
> * and create a new issue for the rest

I created https://github.com/oceanprotocol/df-py/issues/617 for moving the queries.
Should I also create a ticket for casing? Or does https://github.com/oceanprotocol/df-py/issues/505 suffice for now?

Reply:
> Should I also create a ticket for casing? Or does #505 suffice for now?

#505 is just for dftool. Please create a separate issue.

Reply:
Created #618. I will wait for:

> for this issue just move the new predictoor DF-related queries into df-util/predictoor/query.py

in order to mark this issue as resolved.